pkg/cmd/admin/project/new_project.go

```diff
@@ -54,7 +54,12 @@ func NewCmdNewProject(name, fullName string, f *clientcmd.Factory, out io.Writer
 			if options.Client, _, err = f.Clients(); err != nil {
 				kcmdutil.CheckErr(err)
 			}
-			if err := options.Run(); err != nil {
+
+			// We can't depend on len(options.NodeSelector) > 0 as node-selector="" is valid
+			// and we want to populate node selector as project annotation only if explicitly set by user
+			checkNodeSelector := cmd.Flag("node-selector").Changed
```

Comment:
if `--node-selector=` means something different than omitting the option, you should describe the difference in the flag description. (e.g. "If unspecified, the cluster default node selector will be used", etc)